src/main.ts

```diff
@@ -1,46 +1,42 @@
 import 'phaser';
 import player from './assets/player.png';
-import { Player } from './player';
+import { Player, PlayerState } from './player';
+
 type Keys = {
-    jump: Phaser.Input.Keyboard.Key | null,
-    left: Phaser.Input.Keyboard.Key | null,
-    right: Phaser.Input.Keyboard.Key | null,
+    jump: Phaser.Input.Keyboard.Key,
+    left: Phaser.Input.Keyboard.Key,
+    right: Phaser.Input.Keyboard.Key,
 }
 class TestScene extends Phaser.Scene {
-    player: null | Player;
-    ghost: null | Phaser.GameObjects.Sprite;
-    w: number
-    h: number
-    keys: Keys
-    
-    //constants. modify these to tweak 
-    static GRAVITY: number = 50;
-    static X_INC: number = 600;
-    
+    player!: Player;
+    ghost!: Phaser.GameObjects.Sprite;
+    w!: number
+    h!: number
+    keys!: Keys
+    obstacles!: (Phaser.GameObjects.Rectangle & { body: Phaser.Physics.Arcade.Body })[];
+
+
+
     constructor() {
         super('test');
-        this.w = 0;
-        this.h = 0;
-        this.player = null;
-        this.ghost = null;
-        this.keys = {
-            jump: null,
-            left: null,
-            right: null,
-        };
     }
     preload() {
         this.load.image('player', player);
     }
-    addKeys(){
+    addKeys() {
+        this.keys = {
+            jump: this.input.keyboard!.addKey('W'),
+            left: this.input.keyboard!.addKey('A'),
+            right: this.input.keyboard!.addKey('D'),
+        }
         this.keys.jump = this.input.keyboard!.addKey('W');
         this.keys.left = this.input.keyboard!.addKey('A');
         this.keys.right = this.input.keyboard!.addKey('D');
         this.keys.jump?.on('down', () => {
             this.player?.jump();
         })
```

Comment:
Don't need `?` here, use `!` instead.

Instead of `.jump()` on the player, just set a variable `isJumpHeld` to `true` when `jump` is pressed and `false` when it isn't. That way the player state can be entirely contained in the player.